volumes/miovision/sql/table/create_anomalous_ranges.sql

```diff
@@ -1,37 +1,84 @@
--- Make a table to store time ranges with anomalous data
--- TODO:
-/*
-ALTER TABLE scannon.anomalous_ranges SET SCHEMA miovision_api;
-ALTER TABLE nwessel.anomaly_investigation_levels SET SCHEMA miovision_api;
-ALTER TABLE nwessel.anomaly_problem_levels SET SCHEMA miovision_api;
-
-Also need to log changes to all three tables following the procedure documented here:
-https://github.com/Toronto-Big-Data-Innovation-Team/bdit_audit-trigger
-Not sure though if this should reference the `gis` schema or have this logging set 
-up in `miovision_api`.
-*/
-
-CREATE TABLE miovision_api.anomaly_investigation_levels (
-    uid text PRIMARY KEY,
-    description text UNIQUE NOT NULL
+CREATE TABLE miovision_api.anomalous_ranges (
+    intersection_uid smallint,
+    classification_uid smallint,
+    notes text NOT NULL,
+    uid serial PRIMARY KEY,
+    investigation_level text NOT NULL,
+    problem_level text NOT NULL,
+    range_start timestamp,
+    range_end timestamp,
+    CONSTRAINT miovision_qc_pkey PRIMARY KEY (uid),
+    CONSTRAINT miovision_qc_intersection_uid_classification_uid_time_range_key
+    UNIQUE NULLS NOT DISTINCT (intersection_uid, classification_uid, range_start, range_end),
+    CONSTRAINT miovision_qc_classification_uid_fkey FOREIGN KEY (classification_uid)
+    REFERENCES miovision_api.classifications (classification_uid) MATCH SIMPLE
+    ON UPDATE NO ACTION
+    ON DELETE NO ACTION,
+    CONSTRAINT miovision_qc_intersection_uid_fkey FOREIGN KEY (intersection_uid)
+    REFERENCES miovision_api.intersections (intersection_uid) MATCH SIMPLE
+    ON UPDATE NO ACTION
+    ON DELETE NO ACTION,
+    CONSTRAINT miovision_qc_problem_level_fkey FOREIGN KEY (problem_level)
+    REFERENCES miovision_api.anomaly_problem_levels (uid) MATCH SIMPLE
+    ON UPDATE NO ACTION
+    ON DELETE NO ACTION,
+    CONSTRAINT miovision_qc_qa_level_fkey FOREIGN KEY (investigation_level)
+    REFERENCES miovision_api.anomaly_investigation_levels (uid) MATCH SIMPLE
+    ON UPDATE NO ACTION
+    ON DELETE NO ACTION,
+    CONSTRAINT anomalous_ranges_range_order CHECK (
```

Comment:
can we add a `miovision` in front of the constraint name? We will have to add the same checks to other data sources later.